tile_retriever.py

```diff
@@ -0,0 +1,233 @@
+        else:
+            data = data.coarsen(x=2**level, y=2**level, boundary="trim").mean()
+
+        print("Working with", dict(data=data.shape, scale=scale, nroftiles=nroftiles))
+
+        for i in range(nroftiles):
+            for j in range(nroftiles):
+                tile = data[
+                    i * TILE_SIZE : (i + 1) * TILE_SIZE,
+                    j * TILE_SIZE : (j + 1) * TILE_SIZE,
+                    :,
+                ]
+
+                # Skip if the tile has nodata pixels
+                nodata_pixel_count = int(tile.sel(band="B02").isin([NODATA]).sum())
+                if nodata_pixel_count:
+                    continue
+
+                # Cloud and bad pixel percentage based on the SCL band
+                cloudy_pixel_count = int(tile.sel(band="SCL").isin(SCL_FILTER).sum())
+                if cloudy_pixel_count / PIXELS_PER_TILE > BAD_PIXEL_MAX_PERCENTAGE:
+                    continue
+
+                # Append only spectral bands to tiles list
+                tiles.append(tile.sel(band=BANDS[:10]).to_numpy())
+
+                # Track bounds, centroids and scale
+                tile_bounds = numpy.array(
+                    [
+                        stack.attrs["spec"].bounds[0] + i * TILE_SIZE * scale,
+                        stack.attrs["spec"].bounds[1] + j * TILE_SIZE * scale,
+                        stack.attrs["spec"].bounds[0] + (i + 1) * TILE_SIZE * scale,
+                        stack.attrs["spec"].bounds[1] + (j + 1) * TILE_SIZE * scale,
+                    ]
+                )
+                bounds.append(tile_bounds)
+
+                centroids.append(bbox_centroid(tile_bounds, stack.attrs["spec"].epsg))
+
+                scales.append(scale)
+
+    return tiles, bounds, centroids, scales
+
+
+def plot(tile):
+    """
+    Plot a tile in RGB.
+    """
+    from matplotlib import pyplot as plt
+
+    rgb = (
+        numpy.array([tile[:, :, 2], tile[:, :, 1], tile[:, :, 0]]).clip(0, 3000) / 3000
+    )
+    plt.imshow(rgb.transpose(1, 2, 0))
+    plt.show()
+
+
+@click.command()
+@click.option(
+    "--start",
+    required=True,
+    type=str,
+    help="Start date in YYYY-MM-DD",
+)
+@click.option(
+    "--end",
+    required=True,
+    type=str,
+    help="End date in YYYY-MM-DD",
+)
+@click.option(
+    "--lon",
+    required=True,
+    type=float,
+    help="Longitude to find tiles",
+)
+@click.option(
+    "--lat",
+    required=True,
+    type=float,
+    help="Latitude to find tiles",
+)
+def process(start="2020-03-01", end="2020-06-01", lon=-105.78, lat=35.79):
+    """
+    Data preparation pipeline.
+    """
+    print(start, end, lon, lat)
+    items = search(start, end, lon, lat)
+    stack = retrieve(items)
+    tiles, bounds, centroids, scales = tiler(stack)
+    print(f"Storing {len(tiles)} tiles")
+    # TODO: Make this an upload to S3.
+    numpy.savez_compressed(
+        f"/datadisk/clay/{stack.id.to_numpy()[0]}.npz",
```

Comment:
Can we have the option to set the folder path? And also create the folder if it doesn't already exist.